### DISCUSSION TITLE
Some FreeBSD issues

### DISCUSSION THREAD

**Discussion by tinola** [sinamics/ztnet#287] [General]
Hi @sinamics 

Well, a lot of fighting behind me...
Finally the Prisma was upgraded to version 5.8.0 with FreeBSD 14 support. In the `ztnet` folder I had to upgrade `prisma` to the newest version from `5.6.0` to `5.8.0` issuing these 2 commands:

```
npm i --save-dev prisma@latest
npm i @prisma/client@latest
```

After that I can go further, but... I covered another problem with @next/swc-freebsd-x64...
Looking here https://www.npmjs.com/package/@next/swc-freebsd-x64?activeTab=versions we have quite outdated 13.2.4 version, since ztnet providing 14.0.3... This is not available to download, so I think we have to consider to do some condition(s).

Building problems occur on FreeBSD 14, but I've just created VM with older FreeBSD 13.2 to check this out, and problem is the same...
After some investigations I've changed line in `package.json` and `package-lock.json` files form
`"next": "14.0.3",`
to
`"next": "13.2.4",`

The older version was downloaded, but it fails with segmentation fault:

```
> ztnet@0.1.0 build
> next build

info  - Loaded env from /root/ztnet/.env
warn  - You have enabled experimental feature (instrumentationHook) in next.config.mjs.
warn  - Experimental features are not covered by semver, and may cause unexpected or broken application behavior. Use at your own risk.

info  - Skipping linting
info  - Checking validity of types
info  - Using wasm build of next-swc
info  - Creating an optimized production build ..Segmentation fault (core dumped)

```

Could you please check if `ztnet` is building correctly in your virtual environment?...
Thanks!

**Comment by sinamics** [comment]
you never give up 😄, thats awesome.
i will do some testing on my freebsd when i get time and get back to you

**Comment by tinola** [comment]
I'd like to ask politely, have you checked ztnet on FreeBSD?... ;)

**Reply by sinamics** [reply]
sorry, i have not. 
Im working on another project at the moment that needs my full attention.
Will look into it as soon as i find time.

**Reply by tinola** [reply]
Sure, thanks for the information.

**Comment by tinola** [comment]
Meanwhile... During my fight, I found some tip [here](https://github.com/vercel/next.js/issues/59997#issuecomment-1930640518).

I changed line in
`~/ztnet/node_modules/next/dist/build/swc/index.js` from:
`const shouldLoadWasmFallbackFirst = !disableWasmFallback && unsupportedPlatform && useWasmBinary || isWebContainer;`
to:
`shouldLoadWasmFallbackFirst = true` (pff, it smells like dirty hack/workaround...),
then the `next` version from `14.0.3` to `14.1.0` in `package.json` file.
After that ztnet standalone was **successfully built on FreeBSD**.

I really don't know what differences are between @next-swc and @next-swc-wasm editions, but forcing `-wasm` edition did the trick :)
I'm sure you'll find a better solution, then that dirty hack/workaround. Take it easy, don't rush, I know you're working on other projects :)

Cheers!
